app/routes/core.rb

```diff
@@ -21,7 +21,7 @@ class Core < Sinatra::Application
         enable :show_exceptions
       end
 
-      unless settings.test?
+      unless settings.test? || settings.development?
```

Comment:
>  I was definitely getting the 404 page instead of the exceptions until I commented that out.

404 or 500?

> Do you all have a Bugsnag API key locally?

No.

Would you be able to push a (broken) branch that is doing the "wrong" thing?
I'm trying to reproduce the problem you are seeing, with an eye to setting up test coverage for it, and also trying to get to the bottom of why bugsnag is not working.
I can obtain a bugsnag API key.